pyro/infer/mcmc/api.py

```diff
@@ -319,13 +324,14 @@ def __init__(self, kernel, num_samples, warmup_steps=None, initial_params=None,
     def run(self, *args, **kwargs):
         num_samples = [0] * self.num_chains
         z_acc = defaultdict(lambda: [[] for _ in range(self.num_chains)])
-        for x, chain_id in self.sampler.run(*args, **kwargs):
-            if num_samples[chain_id] == self.num_samples:
-                self._diagnostics[chain_id] = x
-            else:
-                num_samples[chain_id] += 1
-                for k, v in x.items():
-                    z_acc[k][chain_id].append(v)
+        with pyro.validation_enabled(not self.disable_validation):
```

Comment:
Thanks for catching this! I intended to change [this line](https://github.com/pyro-ppl/pyro/blob/dev/pyro/infer/mcmc/hmc.py#L299) in the last PR but somehow I forgot to do so.